vars/toolValidate.groovy

```diff
@@ -7,6 +7,8 @@ def call(Map parameters = [:]) {
 
     handlePipelineStepErrors (stepName: 'toolValidate', stepParameters: parameters) {
 
+        echo '[WARNING][toolValidate] This step is deprecated, and it will be removed in future versions. Validation is automatically done inside the steps.'
+
```

Comment:
I have never seen something like this with deprecated code, that's why the code is called deprecated code. The step will be deleted in the next release.

Reply:
IMO there is nothing wrong with deprecation an API and pointing to the new one internally. Benefit: no duplicate code. Prerequisite: new code behaves the same (should be ensured for refactorings anyway).

Nobody knows when a new major release will be available, maybe it takes some time.

I mean I will not insist on a change. Would be cool to have some opinions from colleagues how we should deal with deprecations. In case it is OK for the "developers community" of our project to just keep the old code, that is also fine for me. I would like to remember that it is in general a bad idea to keep coding redundant.

Reply:
I would also suggest to remove duplicate code. So far I would be confident in removing it if we keep the functionality in another place.

Reply:
Ok. I will change it. @OliverNocon could you please review this PR?